### ISSUE TITLE
Crashes on emulator

### ISSUE THREAD

**Issue by Ghedeon** [facebook/flipper#707]
## 🐛 Bug Report
Crashes on emulator. It's getting really difficult to justify Flipper, a debug tool shouldn't crash and block the development process.

## To Reproduce
Build & Run

## Environment
Emulator: Pixel 3 API 28, x86
Flipper: 0.30.1
Soloader: 0.8.0
```
    com.facebook.soloader.SoLoader$WrongAbiError: APK was built for a different platform
        at com.facebook.soloader.SoLoader.loadLibraryBySoName(SoLoader.java:645)
        at com.facebook.soloader.SoLoader.loadLibrary(SoLoader.java:577)
        at com.facebook.soloader.SoLoader.loadLibrary(SoLoader.java:525)
        at com.facebook.soloader.NativeLoaderToSoLoaderDelegate.loadLibrary(NativeLoaderToSoLoaderDelegate.java:24)
        at com.facebook.soloader.nativeloader.NativeLoader.loadLibrary(NativeLoader.java:43)
        at com.facebook.jni.HybridData.<clinit>(HybridData.java:31)
        at com.facebook.flipper.android.FlipperThread.run(FlipperThread.java:25)
     Caused by: java.lang.UnsatisfiedLinkError: couldn't find DSO to load: libfbjni.so caused by: dlopen failed: "/data/data/de.zalon.android.debug/lib-main/libfbjni.so" has unexpected e_machine: 40 (EM_ARM)
        at com.facebook.soloader.SoLoader.doLoadLibraryBySoName(SoLoader.java:789)
        at com.facebook.soloader.SoLoader.loadLibraryBySoName(SoLoader.java:639)
        at com.facebook.soloader.SoLoader.loadLibrary(SoLoader.java:577) 
        at com.facebook.soloader.SoLoader.loadLibrary(SoLoader.java:525) 
        at com.facebook.soloader.NativeLoaderToSoLoaderDelegate.loadLibrary(NativeLoaderToSoLoaderDelegate.java:24) 
        at com.facebook.soloader.nativeloader.NativeLoader.loadLibrary(NativeLoader.java:43) 
        at com.facebook.jni.HybridData.<clinit>(HybridData.java:31) 
        at com.facebook.flipper.android.FlipperThread.run(FlipperThread.java:25) 
E/AndroidRuntime: FATAL EXCEPTION: FlipperConnectionThread
    Process: de.zalon.android.debug, PID: 13538
    java.lang.NoClassDefFoundError: <clinit> failed for class com.facebook.flipper.android.EventBase; see exception in other thread
        at com.facebook.flipper.android.FlipperThread.run(FlipperThread.java:25)
E/CrashLogHandler: Stack trace: java.lang.NoClassDefFoundError: <clinit> failed for class com.facebook.flipper.android.EventBase; see exception in other thread
        at com.facebook.flipper.android.FlipperThread.run(FlipperThread.java:25)
E/CrashLogHandler: Stack trace: com.facebook.soloader.SoLoader$WrongAbiError: APK was built for a different platform
        at com.facebook.soloader.SoLoader.loadLibraryBySoName(SoLoader.java:645)
        at com.facebook.soloader.SoLoader.loadLibrary(SoLoader.java:577)
        at com.facebook.soloader.SoLoader.loadLibrary(SoLoader.java:525)
        at com.facebook.soloader.NativeLoaderToSoLoaderDelegate.loadLibrary(NativeLoaderToSoLoaderDelegate.java:24)
        at com.facebook.soloader.nativeloader.NativeLoader.loadLibrary(NativeLoader.java:43)
        at com.facebook.jni.HybridData.<clinit>(HybridData.java:31)
        at com.facebook.flipper.android.FlipperThread.run(FlipperThread.java:25)
     Caused by: java.lang.UnsatisfiedLinkError: couldn't find DSO to load: libfbjni.so caused by: dlopen failed: "/data/data/de.zalon.android.debug/lib-main/libfbjni.so" has unexpected e_machine: 40 (EM_ARM)
        at com.facebook.soloader.SoLoader.doLoadLibraryBySoName(SoLoader.java:789)
        at com.facebook.soloader.SoLoader.loadLibraryBySoName(SoLoader.java:639)
        at com.facebook.soloader.SoLoader.loadLibrary(SoLoader.java:577) 
        at com.facebook.soloader.SoLoader.loadLibrary(SoLoader.java:525) 
        at com.facebook.soloader.NativeLoaderToSoLoaderDelegate.loadLibrary(NativeLoaderToSoLoaderDelegate.java:24) 
        at com.facebook.soloader.nativeloader.NativeLoader.loadLibrary(NativeLoader.java:43) 
        at com.facebook.jni.HybridData.<clinit>(HybridData.java:31) 
        at com.facebook.flipper.android.FlipperThread.run(FlipperThread.java:25) 
```

**Comment by drbear95** [comment]
Hello, i have the same problem.
Here are my dependencies:
```
debugImplementation 'com.facebook.flipper:flipper:0.30.1'
debugImplementation 'com.facebook.soloader:soloader:0.8.0'
debugImplementation 'com.facebook.flipper:flipper-network-plugin:0.30.1'
```
Tried on emulators with x86_64 and x86 versions.
Are there any chances for fixing this?

**Comment by mweststrate** [comment]
Is this for a React Native project? In that case the version of flipper should be kept lower, as described here: https://fbflipper.com/docs/getting-started.html#android (we are still working on a solution, but it is quite involved)

**Comment by Ghedeon** [comment]
It's a regular native android project, not RN. One note, it works with the same x86 emulator but API 29, instead of 28.

**Comment by mweststrate** [comment]
For now it should be safe to diverge the versions of Flipper itself, and the application plugin (the api has been stable for a very long time). I think the build issue is on @passy's radar.

**Comment by gengjiawen** [comment]
Also @passy , Is there any way I can use higher flipper on React Native.

**Comment by passy** [comment]
@gengjiawen Not easily. We're working on it in https://github.com/facebook/react-native/issues/27565

**Comment by gengjiawen** [comment]
> @gengjiawen Not easily. We're working on it in [facebook/react-native#27565](https://github.com/facebook/react-native/issues/27565)

Is it because react native use fbjni too. Duplicate code since react native include the source code ?

**Comment by gengjiawen** [comment]
Maybe remove fbjni source code from react native repo and use https://github.com/facebookincubator/fbjni as gradle dependency ? Is react native has some hack in it ?

**Comment by passy** [comment]
@gengjiawen Yes, that's exactly what I'm trying to do at the moment. :) Hopefully we'll get this landed soon.

**Comment by bizzguy** [comment]
We had a very similar problem in our Android builds.  When inspecting the APK we noticed that the "so" libraries were not being included in the "lib/****" directories.  It occured when we went for version "0.28.0" to "0.30.0"

We actually had to completely get rid of the cache (ie. delete folder "~/.gradle/cache" and rebuild everything).  I thought deleting the entire cache would cause a really long build time but it was only a few minutes longer.
